components/match_ticker/commons/match_ticker_display_components_new.lua

```diff
@@ -0,0 +1,285 @@
+	local winner = tonumber(match.winner)
+
+	return self.root
+		:node(self:opponent(match.match2opponents[1], winner == 1, true):addClass('team-left'))
+		:node(self:versus())
+		:node(self:opponent(match.match2opponents[2], winner == 2):addClass('team-right'))
+end
+
+---@param opponentData table
+---@param isWinner boolean
+---@param flip boolean?
+---@return Html
+function ScoreBoard:opponent(opponentData, isWinner, flip)
+	local opponent = Opponent.fromMatch2Record(opponentData)
+	---@cast opponent -nil
+	if Opponent.isEmpty(opponent) or Opponent.isTbd(opponent) and opponent.type ~= Opponent.literal then
+		opponent = Opponent.tbd(Opponent.literal)
+	end
+
+	local opponentName = Opponent.toName(opponent)
+	if not opponentName then
+		mw.logObject(opponent, 'Invalid Opponent, Opponent.toName returns nil')
+		opponentName = ''
+	end
+
+	local opponentDisplay = mw.html.create('div')
+		:node(OpponentDisplay.InlineOpponent{
+			opponent = opponent,
+			teamStyle = 'short',
+			flip = flip,
+			showLink = opponentName:gsub('_', ' ') ~= CURRENT_PAGE
+		})
+
+	if isWinner then
+		opponentDisplay:addClass('match-winner')
+	end
+
+	return opponentDisplay
+end
+
+---@return Html
+function ScoreBoard:versus()
+	return mw.html.create('div')
+	:addClass('versus')
+	:node(DefaultMatchTickerDisplayComponents.Versus(self.match):create())
+end
+
+---Display class for the details of a match displayed at the bottom of a match ticker
+---@class NewMatchTickerDetails
+---@operator call(table): NewMatchTickerMatch
+---@field root Html
+---@field hideTournament boolean
+---@field onlyHighlightOnValue string?
+---@field match table
+local Details = Class.new(
+	function(self, args)
+		assert(args.match, 'No Match passed to MatchTickerDetails class')
+		self.root = mw.html.create('div'):addClass('match-details')
+		self.hideTournament = args.hideTournament
+		self.onlyHighlightOnValue = args.onlyHighlightOnValue
+		self.match = args.match
+	end
+)
+
+---@return Html
+function Details:create()
+	local highlightCondition = HighlightConditions.match or HighlightConditions.tournament
+	if highlightCondition(self.match, {onlyHighlightOnValue = self.onlyHighlightOnValue}) then
+		self.root:addClass(HIGHLIGHT_CLASS)
+	end
+
+	return self.root
+		:node(self:streams())
+		:node(self:tournament())
+		:node(self:countdown())
+end
+
+---It will display both countdown and date of the match so the user can select which one to show
+---@return Html
+function Details:countdown()
+	local match = self.match
+
+	local dateString
+	if Logic.readBool(match.dateexact) then
+		local timestamp = DateExt.readTimestamp(match.date) + (Timezone.getOffset(match.extradata.timezoneid) or 0)
+		dateString = DateExt.formatTimestamp('F j, Y - H:i', timestamp) .. ' '
+				.. (Timezone.getTimezoneString(match.extradata.timezoneid) or (Timezone.getTimezoneString('UTC')))
+	else
+		dateString = mw.getContentLanguage():formatDate('F j, Y', match.date) .. (Timezone.getTimezoneString('UTC'))
```

Comment:
There should never be a countdown for non-exact, or am I think wrongly here?

Reply:
The countdown is also responsible for just showing the date in its current state